### PR TITLE
Validate simulation parameters instead of panicking (#10)

### LINKED ISSUE #10: Validate HTTP simulation parameters instead of panicking during conversion
## Summary

The request structs accept unconstrained `f64` and `usize` values, then infallible `From` conversions call the panicking `pos!` macro. Invalid client input can panic a request worker rather than return 400.

## Evidence

As of `348611b`:

- `CreateSessionRequest` and `UpdateSessionRequest` contain raw numeric fields without validation (`src/api/rest/requests.rs:9-49`, `110-154`).
- Conversion uses `pos!` for prices, expirations, volatilities, spreads, model parameters, custom timeframes, and historical prices (`src/session/model.rs:148-173`, `src/api/rest/models.rs:390-519`).
- `pos!` unwraps and panics for negative/non-representable values.
- Decimal conversions use `unwrap_or_default`, silently turning invalid/non-finite values into zero.
- `steps`, `chain_size`, and historical price vector length have no resource bounds.

A regression test converting a request with `initial_price = -1.0` panics with “Value must be positive”.

## Impact

Malformed requests can cause panics, silently altered parameters, invalid model domains, and resource-exhaustion requests.

## Proposed direction

Use fallible `TryFrom` conversions plus request validation. Reject non-finite numbers, invalid ranges/model invariants, zero/oversized steps and chain sizes, and oversized historical payloads with structured 400 responses.

## Acceptance criteria

- No user-controlled numeric value reaches `unwrap`/panic.
- Invalid values return a documented 400 error with field details.
- Reasonable upper bounds are configurable and tested.

## Revalidation

Revalidated on `4b49eca` after the workspace dependency migration; the root cause is still present.

### PR DESCRIPTION
**Pull request by joaquinbejar**
## Summary

Request DTOs accepted unconstrained numbers and the infallible `From`
conversions called panicking macros — a request with `initial_price = -1.0`
panicked a worker instead of returning 400, invalid decimals were silently
zeroed by `unwrap_or_default`, and `steps`/`chain_size`/historical payloads
had no resource bounds. All user-controlled numerics are now validated through
fallible conversions that return a structured 400 naming the offending field.

## Stack

- **Stack:** #15 → #13 → #7 → #10 → #21 → #5 → #4 → #6 → #20 → #19 → #8 → #9 → #11 → #12 → #14 → #17 → #18 → #16 → #22 — **this is PR 4**
- **Base branch:** `stack/3-7-unique-session-ids`
- **Stacked on:** #25 · **Blocks:** the next PR in the stack (#21)
- The diff is cumulative until the lower PRs merge — review against the base branch.

## Changes

- New `ChainError::Validation { field, reason }` → HTTP 400 with `{"error", "field"}` body.
- `From` → `TryFrom` for `CreateSessionRequest → SimulationParameters` and
  `ApiWalkType → WalkType` (still exhaustive, no `_ =>`): finite checks before
  conversion, `Positive::new` instead of `pos_or_panic!` on user input,
  `Decimal::try_from` errors propagate, autocorrelation ∈ [-1,1], historical
  prices each positive and capped.
- `ApiTimeFrame::Custom` validated (finite, > 0) via `time_frame_field` — the
  infallible conversion panicked on a negative custom timeframe.
- Env-configurable bounds in `api/rest/limits.rs` (`OCS_MAX_STEPS` 10_000,
  `OCS_MAX_CHAIN_SIZE` 500, `OCS_MAX_HISTORICAL_PRICES` 100_000; parse-once
  `LazyLock`, warn+default on invalid).
- Shared `api/rest/validation.rs` helpers reused by create/PUT/PATCH merge
  paths; PATCH no longer uses any panicking conversion.
- POST metric increment moved after validation (a rejected request no longer
  inflates the active-sessions gauge — completes under #16 later in the stack).

## Contract impact

Additive: requests that previously panicked the worker or silently zeroed
values now return a documented 400 whose body includes `field`. Success
responses unchanged. Effective-seed generation and the same-seed ⇒ same-tape
contract untouched (reproducibility tests green).

## Testing

- [x] `LOGLEVEL=WARN cargo test --workspace`: 280 + 2 passed, 0 failed (~30 new tests)
- [x] Panic regression: `initial_price = -1.0` → `Validation`, no panic; NaN/inf/zero-steps/over-bounds/oversized-prices/autocorrelation 1.5 all covered
- [x] Bounds parser unit-tested as a pure function; defaults asserted
- [x] clippy `-D warnings` + fmt clean

## Checklist

- [x] f64 → typed conversion stays only in the `From`/`TryFrom` impls + shared validators
- [x] `ApiWalkType` matches remain exhaustive both directions
- [x] No `.unwrap()` / `.expect()` on request input paths
- [x] No new dependencies; no `unsafe`; `tracing` only

Closes #10

### CODE REVIEW

File: src/api/rest/models.rs [hunk 1]
```diff
@@ -500,24 +518,45 @@ impl From<ApiWalkType> for WalkType {
...
+                            prices.len()
+                        ),
+                    });
+                }
+                let mut converted = Vec::with_capacity(prices.len());
+                for price in prices {
+                    converted.push(strictly_positive_field("prices", price)?);
+                }
+                WalkType::Historical {
+                    timeframe: timeframe.into(),
```
Comment (joaquinbejar) [inline src/api/rest/models.rs]: 🔴 **Critical**: this branch bypasses `time_frame_field` and calls the infallible `From<ApiTimeFrame>`, whose `Custom` arm uses `pos_or_panic!`. A request with a negative custom historical timeframe still panics the worker. Convert with `time_frame_field(...)?` here and add regression tests for zero and negative custom historical timeframes.

File: src/api/rest/models.rs [hunk 2]
```diff
@@ -468,13 +486,13 @@ impl From<ApiWalkType> for WalkType {
...
-                theta: pos_or_panic!(theta),
-                xi: pos_or_panic!(xi),
-                rho: Decimal::try_from(rho).unwrap_or_default(),
+                dt: strictly_positive_field("dt", dt)?,
+                drift: decimal_field("drift", drift)?,
+                volatility: positive_field("volatility", volatility)?,
+                kappa: positive_field("kappa", kappa)?,
+                theta: positive_field("theta", theta)?,
+                xi: positive_field("xi", xi)?,
+                rho: decimal_field("rho", rho)?,
```
Comment (joaquinbejar) [inline src/api/rest/models.rs]: 🟠 **Warning**: `rho` is only checked for finiteness here, so values outside `[-1, 1]` are accepted at creation and fail later while simulating. Bound `rho` here like `autocorrelation`; likewise reject GARCH parameters where `alpha + beta >= 1` before storing the session.

File: src/api/rest/handlers.rs [hunk 3]
```diff
@@ -48,7 +48,8 @@ use uuid::Uuid;
     ),
     responses(
         (status = 201, description = "Session created successfully", body = SessionResponse),
-        (status = 400, description = "Invalid request parameters", body = ErrorResponse),
+        (status = 400, description = "Validation failed: a parameter was non-finite, out of range (e.g. negative price/volatility), or steps/chain_size exceeded the configured limits. The body includes an `error` message and the offending `field`.", body = ErrorResponse),
```
Comment (joaquinbejar) [inline src/api/rest/handlers.rs]: 🟠 **Warning**: this annotation says the body includes `field` but declares `ErrorResponse`, whose schema only contains `error`; the PUT and PATCH 400s omit a body schema entirely. Add a validation-error DTO and use it for every validation response so generated clients match the actual wire shape.

**Review by joaquinbejar — COMMENTED** [review]
The broad validation work still leaves one request-path panic and two contract gaps. Intended decision REQUEST_CHANGES, submitted as COMMENT because GitHub does not allow requesting changes on a pull request authored by the current account.

File: src/api/rest/models.rs (hunk 1 above)
Comment (joaquinbejar) [reply src/api/rest/models.rs]: Fixed in ad45316, the Historical arm now converts through time_frame_field and there are regressions for zero and negative custom timeframes.

File: src/api/rest/models.rs (hunk 2 above)
Comment (joaquinbejar) [reply src/api/rest/models.rs]: Done in ad45316, rho goes through bounded_decimal_field on [-1, 1] and GARCH rejects alpha + beta >= 1 at conversion, both with tests.

File: src/api/rest/handlers.rs (hunk 3 above)
Comment (joaquinbejar) [reply src/api/rest/handlers.rs]: Added a ValidationErrorResponse DTO with error and field, map_error emits it, and the POST, PUT, and PATCH 400 annotations now declare it, registered in the OpenAPI components in ad45316. This also carries the PR 24 follow-up, historical symbols are validated at this conversion boundary now.

**Comment by joaquinbejar** [comment]
All three addressed in ad45316 plus the symbol validation promised on PR 24. Rebased onto main.
